File: misc/container_water.py





from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)

        l, r = 0 , n-1


        maxarea = 0
        for i in range(n):
            # move the smaller ptr
            area = min(height[l], height[r]) * abs(l-r)
            maxarea = max(maxarea ,area)
            if height[l] <= height[r]:
                l += 1
            elif height[l] > height[r]:
                r -= 1

        return maxarea


        


        

File: misc/container_water.cpp














#include <bits/stdc++.h>
using namespace std;



class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area = numeric_limits<int>::min();


        int l = 0, r = height.size()-1;

        int s;
        int a;
        while (l < r) {
            s = min(height[l], height[r]);
            a = s * (r-l);
            max_area = max(max_area, a);
            if (height[l] == s) l++;
            else if (height[r] == s) r--;

            
        }
        return max_area;



        
    }
};
